src/v0/destinations/salesforce/utils.js

```diff
@@ -324,6 +325,22 @@ async function getSalesforceIdForRecordUsingSdk(
       `SELECT Id FROM ${objectType} WHERE ${identifierType} = '${identifierValue}'`,
     );
   } catch (error) {
+    // check if the error message contains 'session expired'
+    if (error.message?.toLowerCase().includes('session expired')) {
```

Comment:
If we keep the session-expired check in every try/catch where the SDK is invoked, it’s easy to miss it when a new call is added. It’s safer to handle this in a single place when the SDK is initialized.